### PR TITLE
new port: devel/py-decorator4

### PR DESCRIPTION
**Pull request by yzgyyang**
devel/py-decorator needs to be updated to 5.x, while some ports don't
support that in their latest releases, including:

- devel/py-pytest-relaxed
- multimedia/py-moviepy

Updated them to use this devel/py-decorator4 until they officially
support 5.x. No functional changes for these ports.

Approved by:	lwhsu (mentor), portmgr (implicit, no functional change)

### CODE REVIEW

File: devel/py-decorator4/Makefile
Comment (lwhsu) [inline devel/py-decorator4/Makefile:17]: `3.6+` is not really needed, but since it's a copy so it's also fine to keep it.

**Comment by lwhsu** [comment]
> Approved by: lwhsu (mentor), portmgr (implicit, no functional change)

Which part of the patch needs portmgr check?

File: devel/py-decorator4/Makefile
Comment (yzgyyang) [reply devel/py-decorator4/Makefile:17]: Noted, I thought the same as well so kept them identical.

**Comment by yzgyyang** [comment]
> > Approved by: lwhsu (mentor), portmgr (implicit, no functional change)
> 
> Which part of the patch needs portmgr check?

I also updated the two ports mentioned without seeking for maintainer approval - since they are cosmetic changes. I am not sure how to best indicate that in the commit message, so the portmgr part is an attempt. ;)

What do you think? @lwhsu

**Comment by yzgyyang** [comment]
Also running build tests for the affected ports:

- devel/py-pytest-relaxed
- multimedia/py-moviepy

https://poudriere.ygy.io/build.html?mastername=130Ramd64-pub&build=2021-05-19_07h52m14s

**Comment by lwhsu** [comment]
BTW when adding 

> > > Approved by: lwhsu (mentor), portmgr (implicit, no functional change)
> > 
> > 
> > Which part of the patch needs portmgr check?
> 
> I also updated the two ports mentioned without seeking for maintainer approval - since they are cosmetic changes. I am not sure how to best indicate that in the commit message, so the portmgr part is an attempt. ;)
> 
> What do you think? @lwhsu

Oh, chasing the dependency update is an (undocumented? I guess) allowed update. So no worries.

**Comment by yzgyyang** [comment]
@lwhsu Do I need to remove the portmgr part, or commit as-is is fine?

**Review by lwhsu — APPROVED** [review]
Looks good to me. BTW, recently I am thinking that adding this "keeping old version" port should come with a DEPRECATED date, which can be set to a long date in the future, for remind us to revise it when the date approaches and not letting the ports tree have too many old but nobody knows it. It should also safe from being deleted when DEPRECATED date comes but there are still users because one should check the dependency when deleting a port, and we can of course extend the date if it needs.

**Comment by lwhsu** [comment]
> @lwhsu Do I need to remove the portmgr part, or commit as-is is fine?

I feel that it's not necessary to include that.

**Comment by yzgyyang** [comment]
> > @lwhsu Do I need to remove the portmgr part, or commit as-is is fine?
> 
> I feel that it's not necessary to include that.

Thanks! I have removed that part.

**Comment by yzgyyang** [comment]
> Looks good to me. BTW, recently I am thinking that adding this "keeping old version" port should come with a DEPRECATED date, which can be set to a long date in the future, for remind us to revise it when the date approaches and not letting the ports tree have too many old but nobody knows it. It should also safe from being deleted when DEPRECATED date comes but there are still users because one should check the dependency when deleting a port, and we can of course extend the date if it needs.

This is a great idea! I updated the PR and the commit message - please reapprove. ;)

**Review by lwhsu — APPROVED** [review]

**Comment by yzgyyang** [comment]
@lwhsu For context, I fixed the issue mentioned by koobs at https://github.com/yzgyyang/freebsd-ports/commit/4d4cadcf9d489d5e78e160190c92f2c8bc44d307, but later wrongly pushed an older branch and committed that one instead.

**Pull request merged**
